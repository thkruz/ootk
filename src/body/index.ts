export { Celestial } from './Celestial';
export { Earth } from './Earth';
export { Moon } from './Moon';
export { Sun } from './Sun';
