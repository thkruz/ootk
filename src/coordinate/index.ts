export * from './Hill.js';
export * from './ClassicalElements.js';
export * from './EquinoctialElements.js';
export * from './FormatTle.js';
export * from './Geodetic.js';
export * from './ITRF.js';
export * from './J2000.js';
export * from './RelativeState.js';
export * from './RIC.js';
export * from './StateVector.js';
export * from './TEME.js';
export * from './Tle.js';
