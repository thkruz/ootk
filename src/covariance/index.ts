export * from './CovarianceSample.js';
export * from './StateCovariance.js';
