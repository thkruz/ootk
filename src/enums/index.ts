export { CatalogSource } from './CatalogSource.js';
export { CommLink } from './CommLink.js';
export { OrbitRegime } from './OrbitRegime.js';
export { Sgp4OpsMode } from './Sgp4OpsMode.js';
export { AngularDiameterMethod } from './AngularDiameterMethod.js';
export { AngularDistanceMethod } from './AngularDistanceMethod.js';
export { PassType } from './PassType.js';
