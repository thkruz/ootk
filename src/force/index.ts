export { EarthGravity } from './EarthGravity.js';
export { Force } from './Force.js';
export { ForceModel } from './ForceModel.js';
export { Gravity } from './Gravity.js';
export { SolarRadiationPressure } from './SolarRadiationPressure.js';
export { ThirdBodyGravity } from './ThirdBodyGravity.js';
export { Thrust } from './Thrust.js';

