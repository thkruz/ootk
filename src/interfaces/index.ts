export type { GroundPositionParams } from './GroundPositionParams.js';
export type { OptionsParams } from './OptionsParams.js';
export type { TimeVariables } from './TimeVariables.js';
export type { StarObjectParams } from './StarObjectParams.js';
export type { ClassicalElementsParams } from './ClassicalElementsParams.js';
export type { EquinoctialElementsParams } from './EquinoctialElementsParams.js';
export type { ScenarioParams } from '../objects/Scenario.js';
export type { BaseObjectParams } from '../objects/base/BaseObject.js';
export type { GroundObjectParams } from '../objects/base/GroundObject.js';
export type { FacilityParams } from '../objects/base/Facility.js';
export type { SensorParams } from '../objects/attachable/SensorAdv.js';
export type { RectangularSensorParams } from '../objects/attachable/RectangularSensor.js';
