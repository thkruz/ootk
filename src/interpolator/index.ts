export { ChebyshevCoefficients } from './ChebyshevCoefficients.js';
export { ChebyshevInterpolator } from './ChebyshevInterpolator.js';
export { CubicSpline } from './CubicSpline.js';
export { CubicSplineInterpolator } from './CubicSplineInterpolator.js';
export { Interpolator } from './Interpolator.js';
export { LagrangeInterpolator } from './LagrangeInterpolator.js';
export { StateInterpolator } from './StateInterpolator.js';
export { VerletBlendInterpolator } from './VerletBlendInterpolator.js';

