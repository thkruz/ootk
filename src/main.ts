/**
 * @author @thkruz Theodore Kruczek
 * @description Orbital Object ToolKit (ootk) is a collection of tools for working
 * with satellites and other orbital objects.
 * @license AGPL-3.0-or-later
 * @copyright (c) 2020-2024 Theodore Kruczek
 *
 * Many of the classes are based off of the work of @david-rc-dayton and his
 * Pious Squid library (https://github.com/david-rc-dayton/pious_squid).
 *
 * Orbital Object ToolKit is free software: you can redistribute it and/or modify it under the
 * terms of the GNU Affero General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later version.
 *
 * Orbital Object ToolKit is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License along with
 * Orbital Object ToolKit. If not, see <http://www.gnu.org/licenses/>.
 */

export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './time/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
export * from './operations/operations.js';
export { BaseObject, Sensor } from './objects/index.js';
export { Earth } from './body/index.js';
export * from './coordinate/index.js';
export * from './observation/index.js';
export * from './data/DataHandler.js';
export * from './sgp4/index.js';
export * from './objects/index.js';
export * from './interpolator/index.js';
export * from './force/index.js';
export * from './maneuver/index.js';
export * from './optimize/index.js';
export * from './orbit_determination/index.js';
export * from './body/index.js';
export * from './enums/index.js';
export * from './types/types.js';
export * from './interfaces/index.js';
export * from './coordinate/index.js';
export * from './operations/index.js';
export * from './force/index.js';
export * from './propagator/index.js';
export * from './transforms/index.js';
export * from './utils/index.js';
