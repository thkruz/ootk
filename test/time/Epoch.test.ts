import { Epoch, Seconds } from '../../src/main';

describe('Epoch', () => {
  it('should be constructed from a POSIX timestamp', () => {
    const posix = 1614556800 as Seconds;
    const epoch = new Epoch(posix);

    expect(epoch.posix).toEqual(posix);
  });

  // toString
  it('should convert to a string', () => {
    const posix = 1614556800 as Seconds;
    const epoch = new Epoch(posix);

    expect(epoch.toString()).toMatchSnapshot();
  });

  // toExcelString
  it('should convert to an Excel string', () => {
    const posix = 1614556800 as Seconds;
    const epoch = new Epoch(posix);

    expect(epoch.toExcelString()).toMatchSnapshot();
  });

  // difference
  it('should calculate the difference between two epochs', () => {
    const posix1 = 1614556800 as Seconds;
    const posix2 = (1614556800 + 60) as Seconds;
    const epoch1 = new Epoch(posix1);
    const epoch2 = new Epoch(posix2);

    expect(epoch1.difference(epoch2)).toEqual(-60);
  });

  // equals
  it('should check if two epochs are equal', () => {
    const posix1 = 1614556800 as Seconds;
    const posix2 = (1614556800 + 60) as Seconds;
    const epoch1 = new Epoch(posix1);
    const epoch2 = new Epoch(posix2);

    expect(epoch1.equals(epoch2)).toEqual(false);
  });

  // toDateTime
  it('should convert to a DateTime object', () => {
    const posix = 1614556800 as Seconds;
    const epoch = new Epoch(posix);

    expect(epoch.toDateTime()).toMatchSnapshot();
  });

  // toEpochYearAndDay
  it('should convert to an epoch year and day', () => {
    const posix = 1614556800 as Seconds;
    const epoch = new Epoch(posix);

    expect(epoch.toEpochYearAndDay()).toMatchSnapshot();
  });

  // toJulianDate
  it('should convert to a Julian date', () => {
    const posix = 1614556800 as Seconds;
    const epoch = new Epoch(posix);

    expect(epoch.toJulianDate()).toMatchSnapshot();
  });

  // toJulianCenturies
  it('should convert to Julian centuries', () => {
    const posix = 1614556800 as Seconds;
    const epoch = new Epoch(posix);

    expect(epoch.toJulianCenturies()).toMatchSnapshot();
  });

  // operatorGreaterThan
  it('should check if one epoch is greater than another', () => {
    const posix1 = 1614556800 as Seconds;
    const posix2 = (1614556800 + 60) as Seconds;
    const epoch1 = new Epoch(posix1);
    const epoch2 = new Epoch(posix2);

    expect(epoch1.operatorGreaterThan(epoch2)).toEqual(false);
  });

  // operatorGreaterThanOrEqual
  it('should check if one epoch is greater than or equal to another', () => {
    const posix1 = 1614556800 as Seconds;
    const posix2 = (1614556800 + 60) as Seconds;
    const epoch1 = new Epoch(posix1);
    const epoch2 = new Epoch(posix2);

    expect(epoch1.operatorGreaterThanOrEqual(epoch2)).toEqual(false);
  });

  // operatorLessThan
  it('should check if one epoch is less than another', () => {
    const posix1 = 1614556800 as Seconds;
    const posix2 = (1614556800 + 60) as Seconds;
    const epoch1 = new Epoch(posix1);
    const epoch2 = new Epoch(posix2);

    expect(epoch1.operatorLessThan(epoch2)).toEqual(true);
  });

  // operatorLessThanOrEqual
  it('should check if one epoch is less than or equal to another', () => {
    const posix1 = 1614556800 as Seconds;
    const posix2 = (1614556800 + 60) as Seconds;
    const epoch1 = new Epoch(posix1);
    const epoch2 = new Epoch(posix2);

    expect(epoch1.operatorLessThanOrEqual(epoch2)).toEqual(true);
  });
});
