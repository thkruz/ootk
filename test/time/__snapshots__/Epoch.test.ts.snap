// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Epoch should convert to Julian centuries 1`] = `0.21162217659137578`;

exports[`Epoch should convert to a DateTime object 1`] = `2021-03-01T00:00:00.000Z`;

exports[`Epoch should convert to a Julian date 1`] = `2459274.5`;

exports[`Epoch should convert to a string 1`] = `"2021-03-01T00:00:00.000Z"`;

exports[`Epoch should convert to an Excel string 1`] = `"2021-03-01T00:00:00"`;

exports[`Epoch should convert to an epoch year and day 1`] = `
Object {
  "epochDay": "060.00000000",
  "epochYr": "21",
}
`;
